python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -40,19 +40,68 @@
 
 
 class ResourceDemandScheduler:
-    def __init__(self, provider: NodeProvider,
+    def __init__(self,
+                 provider: NodeProvider,
                  node_types: Dict[NodeType, NodeTypeConfigDict],
-                 max_workers: int):
+                 max_workers: int,
+                 aggressive: bool = False,
+                 target_utilization_fraction: float = 1):
         self.provider = provider
         self.node_types = copy.deepcopy(node_types)
         self.max_workers = max_workers
-        # is_legacy_yaml tracks if the cluster configs was originally without
-        # available_node_types and was autofilled with available_node_types.
-        self.is_legacy_yaml = (NODE_TYPE_LEGACY_HEAD in node_types
-                               and NODE_TYPE_LEGACY_WORKER in node_types)
+        self.aggressive = aggressive
```

Comment:
Aggressive / legacy util fraction are legacy concepts, how about we rewrite them to the upscaling_fraction / initial_num_nodes constants in the autoscaler so these concepts aren't ever in ResourceDemandScheduler?

Reply:
Can we just use upscaling_fraction instead of both?